front/app/containers/IdeasNewPage/WithJSONForm/index.tsx

```diff
@@ -113,10 +115,23 @@ const IdeasNewPageWithJSONForm = ({ params }: WithRouterProps) => {
     });
     const ideaId = idea.data.id;
 
-    clHistory.push({
-      pathname: `/ideas/${idea.data.attributes.slug}`,
-      search: `?new_idea_id=${ideaId}`,
-    });
+    // Check ParticipationMethodConfig for form submission action
+    if (!isNilOrError(phases)) {
+      const phaseParticipationMethod =
+        getLatestRelevantPhase(phases)?.attributes.participation_method;
+      if (!isNilOrError(phaseParticipationMethod)) {
+        getMethodConfig(phaseParticipationMethod).onFormSubmission(
+          project,
+          ideaId,
+          idea
+        );
+      }
+    }
+    if (!isNilOrError(project)) {
+      getMethodConfig(
+        project?.attributes.participation_method
+      ).onFormSubmission(project, ideaId, idea);
```

Comment:
Should I rename the idea and ideaId  to "input" and "inputId" now? Or wait until later when we do a larger refactor?

Reply:
I would say: do what is most appropriate for you. I tend to rename stuff when I make changes to a file. Positive effect: gradually things have the right name. Negative effect: PRs become bigger and make the goal of the PR less clear.